### PR TITLE
Add reconciliation endpoint

### PR DESCRIPTION
**Pull request by aliveni**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/BPS-1359

### Change description ###
Add reconciliation endpoint which accepts the supplier statements report and saves in database.

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: src/integrationTest/java/uk/gov/hmcts/reform/blobrouter/controllers/ReconciliationControllerDisabledTest.java [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+        mockMvc
+            .perform(
+                post("/reform-scan/reconciliation-report/10082020")
+                    .header(HttpHeaders.CONTENT_TYPE, MediaType.APPLICATION_JSON_VALUE)
+                    .content(requestBody)
+            )
+            .andDo(print())
+            .andExpect(status().isNotFound());
+    }
+}
```
Comment (doncem) [inline src/integrationTest/java/uk/gov/hmcts/reform/blobrouter/controllers/ReconciliationControllerDisabledTest.java:45]: perhaps no need to auto configure mvc. just stick in app context and verify controller bean is not loaded :wink:

File: src/integrationTest/java/uk/gov/hmcts/reform/blobrouter/controllers/ReconciliationControllerTest.java [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+        mockMvc
+            .perform(
+                post("/reform-scan/reconciliation-report/2020-08-10")
+                    .header(HttpHeaders.CONTENT_TYPE, MediaType.APPLICATION_JSON_VALUE)
+                    .content(requestBody)
+            )
+            .andDo(print())
+            .andExpect(status().isOk())
+            .andExpect(MockMvcResultMatchers.jsonPath("id").exists())
+            .andExpect(MockMvcResultMatchers.jsonPath("id").isNotEmpty());
```
Comment (doncem) [inline src/integrationTest/java/uk/gov/hmcts/reform/blobrouter/controllers/ReconciliationControllerTest.java:44]: `isNotEmpty` should be enough? it's a test. if it fails - it fails

File: src/main/java/uk/gov/hmcts/reform/blobrouter/reconciliation/controller/ReconciliationController.java [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+public class ReconciliationController {
+
+    private final ReconciliationService service;
+
+    public ReconciliationController(ReconciliationService service) {
+        this.service = service;
+    }
+
+    @PostMapping(
+        path = "/reform-scan/reconciliation-report/{date}",
```
Comment (doncem) [inline src/main/java/uk/gov/hmcts/reform/blobrouter/reconciliation/controller/ReconciliationController.java:32]: `/reform-scan` prefix is intended? asking just for clarification if it is

Comment (aliveni) [reply src/main/java/uk/gov/hmcts/reform/blobrouter/reconciliation/controller/ReconciliationController.java:32]: the swagger spec has the path as `/reform-scan/reconciliation-report/{date}`

File: src/integrationTest/java/uk/gov/hmcts/reform/blobrouter/controllers/ReconciliationControllerDisabledTest.java [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+        // given
+        String requestBody = Resources.toString(
+            getResource("reconciliation/valid-supplier-statement-report.json"),
+            UTF_8
+        );
+
+        // when
+        mockMvc
+            .perform(
+                post("/reform-scan/reconciliation-report/10082020")
```
Comment (rkondratowicz) [inline src/integrationTest/java/uk/gov/hmcts/reform/blobrouter/controllers/ReconciliationControllerDisabledTest.java:38]: invalid date format

**Pull request merged**